controllers/agentcluster_controller.go

```diff
@@ -137,6 +138,12 @@ func (r *AgentClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	err = r.ensureClusterDeploymentAnnotations(ctx, clusterDeployment)
```

Comment:
I can't tell from the context, but we need to make sure that this is conditional based on the type of cluster. You *do* want hive to reconcile non-HCP clusters!